### ISSUE TITLE
Broken integration on Windows, Firefox Nightly (2023-04-26)

### ISSUE THREAD

**Issue by StoyanDimitrov** [keepassxreboot/keepassxc-browser#1905]
After browser update the integration stopped working with "Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings." in the plug-in pop-up and if I click on the button I get "Key exchange was not successful."

## Expected Behavior
The integration should continue to work. Well it does work on Tor 12.0.5.

## Current Behavior
Error is shown.

## Possible Solution
The browser console reveals the following message:

> `OperationError: Could not parse path (C:/Users/username/AppData/Local/KeePassXC/org.keepassxc.keepassxc_browser_tor-browser.json): NS_ERROR_FILE_UNRECOGNIZED_PATH`

This path is basically invalid on Windows. Correcting it in the registry fixes the issue and the integration is again functional.

## Debug info

KeePassXC - 2.7.4
KeePassXC-Browser - 1.8.6.1
Operating system: Windows
Browser: Firefox, Tor

**Comment by droidmonkey** [comment]
We don't support problems in nightly or beta builds.

**Comment by pallaswept** [comment]
I have filed a bug with mozilla, adding your voice there may help this to get attention from them: https://bugzilla.mozilla.org/show_bug.cgi?id=1830605

**Comment by pallaswept** [comment]
> We don't support problems in nightly or beta builds.

Of course you can't support problems in the nightly/beta builds, but my experience with Mozilla has taught me that if you let them know about the bug while they're still actively working on the thing that caused it, then they usually fix it quite quickly, and the bug never gets past a new nightly builds. If you wait until the bug passes nightly, into beta, or worse yet, into the production build, then getting the bug fixed has to get attention of developers who are no longer working on the relevant code, which can take months, then the fix has to pass through nightly and beta, taking months more, before making it to the final build.

TL;DR you can't support broken betas, but it is easier to squash new bugs as soon as you see them, before they get big and strong.

And of course, bug reports from actual developers of important software, such as yourself, are taken more seriously than reports from just some random person like one of us, so if you could add your voice to the bugzilla issue I filed up there, it would be really helpful for us all.

Thanks :)

**Comment by pallaswept** [comment]
> This path is basically invalid on Windows. Correcting it in the registry fixes the issue and the integration is again functional.

To expand on this, I've been able to fix this by altering this registry key also. Thanks for the tip, OP!

The key in question is `HKEY_CURRENT_USER\Software\Mozilla\NativeMessagingHosts\`
The thing that makes it invalid is the slashes, so to fix it, you want to change all the forward-slash characters into back-slash characters, ie change the value from 
`C:/Users/username/AppData/Local/KeePassXC/org.keepassxc.keepassxc_browser_tor-browser.json`
to
`C:\Users\username\AppData\Local\KeePassXC\org.keepassxc.keepassxc_browser_tor-browser.json`

And now everything works again. Hope this helps anyone else trying to fix it, and I'll pass this on to bugzilla.
